src/extensions/renderer/base/node-shapes.js

```diff
@@ -348,6 +348,104 @@ BRp.generateBarrel = function(){
   } );
 };
 
+BRp.generateBottomRoundrectangle = function(){
+  return ( this.nodeShapes['bottomroundrectangle'] = {
+    renderer: this,
+
+    name: 'bottomroundrectangle',
+
+    points: math.generateUnitNgonPointsFitToSquare( 4, 0 ),
+
+    draw: function( context, centerX, centerY, width, height ){
+      this.renderer.nodeShapeImpl( this.name, context, centerX, centerY, width, height );
+    },
+
+    intersectLine: function( nodeX, nodeY, width, height, x, y, padding ){
+      var topStartX = nodeX - ( width / 2 + padding );
+      var topStartY = nodeY - ( height / 2 + padding );
+      var topEndY = topStartY;
+      var topEndX = nodeX + ( width / 2 + padding );
+
+      var topIntersections = math.finiteLinesIntersect(
+        x, y, nodeX, nodeY, topStartX, topStartY, topEndX, topEndY, false );
+      if( topIntersections.length > 0 ){
+        return topIntersections;
+      }
+
+      return math.roundRectangleIntersectLine(
+        x, y,
+        nodeX,
+        nodeY,
+        width, height,
+        padding )
+      ;
+    },
+
+    // Looks like the width passed into this function is actually the total width / 2
+    checkPoint: function(
+      x, y, padding, width, height, centerX, centerY ){
+
+      var cornerRadius = math.getRoundRectangleRadius( width, height );
+
+      // Check hBox
+      if( math.pointInsidePolygon( x, y, this.points,
+        centerX, centerY, width, height - 2 * cornerRadius, [0, -1], padding ) ){
+        return true;
+      }
+
+      // Check vBox
+      if( math.pointInsidePolygon( x, y, this.points,
+        centerX, centerY, width - 2 * cornerRadius, height, [0, -1], padding ) ){
+        return true;
+      }
+
+      // check non-rounded top side
+      var outerWidth = ( ( width / 2 ) + 2 * padding );
+      var outerHeight = ( ( height / 2 ) + 2 * padding );
+      var points = [
+        centerX - outerWidth, centerY - outerHeight,
+        centerX - outerWidth, centerY,
+        centerX + outerWidth, centerY,
+        centerX + outerWidth, centerY - outerHeight
+      ];
+      if( math.pointInsidePolygonPoints( x, y, points)){
+        return true;
+      }
+
+
+      var checkInEllipse = function( x, y, centerX, centerY, width, height, padding ){
```

Comment:
Is is possible to use the same function as the roundrect or could this function be pulled out to `math` and possibly have the roundrect function use `checkInEllipse`?